NPB3.3-MPI/make.def

```diff
@@ -0,0 +1,162 @@
+#---------------------------------------------------------------------------
+#
+#                SITE- AND/OR PLATFORM-SPECIFIC DEFINITIONS. 
+#
+#---------------------------------------------------------------------------
+
+#---------------------------------------------------------------------------
+# Items in this file will need to be changed for each platform.
+#---------------------------------------------------------------------------
+
+#---------------------------------------------------------------------------
+# Parallel Fortran:
+#
+# For CG, EP, FT, MG, LU, SP and BT, which are in Fortran, the following must 
+# be defined:
+#
+# MPIF77     - Fortran compiler
+# FFLAGS     - Fortran compilation arguments
+# FMPI_INC   - any -I arguments required for compiling MPI/Fortran 
+# FLINK      - Fortran linker
+# FLINKFLAGS - Fortran linker arguments
+# FMPI_LIB   - any -L and -l arguments required for linking MPI/Fortran 
+# 
+# compilations are done with $(MPIF77) $(FMPI_INC) $(FFLAGS) or
+#                            $(MPIF77) $(FFLAGS)
+# linking is done with       $(FLINK) $(FMPI_LIB) $(FLINKFLAGS)
+#---------------------------------------------------------------------------
+
+#---------------------------------------------------------------------------
+# This is the fortran compiler used for MPI programs
+#---------------------------------------------------------------------------
+MPIF77 = mpif77
+# This links MPI fortran programs; usually the same as ${MPIF77}
+FLINK	= $(MPIF77)
+
+#---------------------------------------------------------------------------
+# These macros are passed to the linker to help link with MPI correctly
+#---------------------------------------------------------------------------
+FMPI_LIB  = -L/usr/lib -lmpi
+
+#---------------------------------------------------------------------------
+# These macros are passed to the compiler to help find 'mpif.h'
+#---------------------------------------------------------------------------
+FMPI_INC = -I/usr/include/mpi
+
+#---------------------------------------------------------------------------
+# Global *compile time* flags for Fortran programs
+#---------------------------------------------------------------------------
+FFLAGS	= -O3 -fdefault-integer-8 -mcmodel=large
```

Comment:
You're right, it's better to use the -03 flag for compiling optimizations.

I want to use the default flag provided by default when you download NPB.

> Le 30 déc. 2015 à 17:05, Bop <notifications@github.com> a écrit :
> 
> In NPB3.3-MPI/make.def:
> 
> > +FLINK	= $(MPIF77)
> > +
> > +#---------------------------------------------------------------------------
> > +# These macros are passed to the linker to help link with MPI correctly
> > +#---------------------------------------------------------------------------
> > +FMPI_LIB  = -L/usr/lib -lmpi
> > +
> > +#---------------------------------------------------------------------------
> > +# These macros are passed to the compiler to help find 'mpif.h'
> > +#---------------------------------------------------------------------------
> > +FMPI_INC = -I/usr/include/mpi
> > +
> > +#---------------------------------------------------------------------------
> > +# Global *compile time* flags for Fortran programs
> > +#---------------------------------------------------------------------------
> > +FFLAGS	= -O3 -fdefault-integer-8 -mcmodel=large
> According this page :
> 
> https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html
> 
> I thougth it was better to use the arg -O3 to increases both compilation time and the performance of the generated code.
> 
> -O3 is supposed to be better than -O.
> But -O3 took certainly more place during the compilation.
> 
> Is there another thing which make -O better than -O3 for our use ?
> 
> —
> Reply to this email directly or view it on GitHub.
>